libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -465,7 +465,11 @@ def _create_message_dicts(
     def _create_chat_result(self, response: Union[dict, BaseModel]) -> ChatResult:
         generations = []
         if not isinstance(response, dict):
-            response = response.dict()
+            response = (
+                getattr(response, "model_dump")()
+                if hasattr(response, "model_dump")
+                else response.dict()
+            )
```

Comment:
think we can just model_dump here

Reply:
`response` is expected to have a type of `pydantic.v1.BaseModel`, which has no `model_dump()` function. So we either have to change the type safety or go with the approach i commited (same case [here](https://github.com/langchain-ai/langchain/blob/a6d901a91236b79f74f2ba66ba2444594865916e/libs/partners/openai/langchain_openai/chat_models/azure.py#L214-L218))

Reply:
We could import `BaseModel` from `openai` which supports `model_dump()` no matter the pydantic version, as you suggested [in the previous PR](https://github.com/langchain-ai/langchain/pull/16629#issuecomment-1935081971). It will make the code cleaner